### PR TITLE
otelzap: Add Benchmarks

### PR DESCRIPTION
**Pull request by khushijain21**
Part of #5191

Pre-work #5279

```
goos: linux
goarch: amd64
pkg: go.opentelemetry.io/contrib/bridges/otelzap
cpu: 12th Gen Intel(R) Core(TM) i5-1245U
BenchmarkCoreWrite/10_fields-12         	 1000000	      1116 ns/op	     978 B/op	      13 allocs/op
BenchmarkCoreWrite/20_fields-12         	  407290	      2615 ns/op	    2180 B/op	      22 allocs/op
BenchmarkCoreWrite/Namespace-12         	  725268	      1671 ns/op	    1760 B/op	      16 allocs/op
BenchmarkCoreWrite/With10_fields-12     	 4964038	       260.2 ns/op	     208 B/op	       1 allocs/op
BenchmarkCoreWrite/With20_fields-12     	 2004085	       628.7 ns/op	     640 B/op	       1 allocs/op
BenchmarkCoreWrite/WithNamespace-12     	100000000	        16.95 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	go.opentelemetry.io/contrib/bridges/otelzap	8.619s
```

### CODE REVIEW

**Review by dmathieu — COMMENTED** [review]
Could you update the PR description with the benchmark results, for reference?

File: bridges/otelzap/core_test.go [hunk 1]
```diff
@@ -22,6 +24,10 @@ var (
 	loggerName  = "name"
 	testKey     = "key"
 	testValue   = "value"
+	testentry   = zapcore.Entry{
```
Comment (pellared) [inline bridges/otelzap/core_test.go:27]:
```suggestion
	testEntry   = zapcore.Entry{
```

File: bridges/otelzap/core_test.go [hunk 2]
```diff
@@ -174,3 +180,80 @@ func TestConvertLevel(t *testing.T) {
 		}
 	}
 }
+
+// Benchmark with multiple Fields.
+func BenchmarkMultipleFields(b *testing.B) {
```
Comment (pellared) [inline bridges/otelzap/core_test.go:185]: This is mainly about testing the `Write` method

```suggestion
func BenchmarkCoreWrite(b *testing.B) {
```

File: bridges/otelzap/core_test.go [hunk 3]
```diff
@@ -174,3 +180,80 @@ func TestConvertLevel(t *testing.T) {
...
+	benchmarks := []struct {
+		name  string
+		field []zapcore.Field
```
Comment (pellared) [inline bridges/otelzap/core_test.go:188]:
```suggestion
		fields []zapcore.Field
```

File: bridges/otelzap/core_test.go [hunk 4]
```diff
@@ -174,3 +180,80 @@ func TestConvertLevel(t *testing.T) {
...
+				zap.Time("k", time.Unix(1000, 1000)),
+				zap.String("k", "a"),
+				zap.Ints("k", []int{1, 2}),
+				zap.Object("k", loggable{true}),
+			},
+		},
+	}
+
+	for _, bm := range benchmarks {
+		b.Run(fmt.Sprint("Core Write", bm.name), func(b *testing.B) {
```
Comment (pellared) [inline bridges/otelzap/core_test.go:230]:
```suggestion
		b.Run(bm.name, func(b *testing.B) {
```

File: bridges/otelzap/core_test.go [hunk 5]
```diff
@@ -174,3 +180,80 @@ func TestConvertLevel(t *testing.T) {
...
+					if err != nil {
+						b.Errorf("Unexpected error: %v", err)
+					}
+				}
+			})
+		})
+	}
+
+	for _, bm := range benchmarks {
+		b.Run(fmt.Sprint("Core Write Accumulated fields", bm.name), func(b *testing.B) {
```
Comment (pellared) [inline bridges/otelzap/core_test.go:245]:
```suggestion
		b.Run(fmt.Sprint("With", bm.name), func(b *testing.B) {
```

File: bridges/otelzap/core_test.go (hunk 2 above)
Comment (pellared) [inline bridges/otelzap/core_test.go:185]: Can we also give a benchmark case for namespace handling?

**Review by pellared — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

File: bridges/otelzap/core_test.go [hunk 6]
```diff
@@ -178,3 +184,99 @@ func TestConvertLevel(t *testing.T) {
...
+			name: "10 fields",
+			fields: []zapcore.Field{
+				zap.Int16("a", 1),
+				zap.String("k", "a"),
+				zap.Bool("k", true),
+				zap.Time("k", time.Unix(1000, 1000)),
+				zap.Binary("k", []byte{1, 2}),
+				zap.ByteString("k", []byte{1, 2}),
+				zap.Object("k", loggable{true}),
+				zap.Array("k", loggable{true}),
```
Comment (khushijain21) [inline bridges/otelzap/core_test.go:203]: Added 'Array` as well, this wasn't part of the previous commit

**Pull request merged**